### PR TITLE
[du] add backoff for duckdb connections

### PR DESCRIPTION
**Pull request by dehume**
## Summary & Motivation

Update code snippets for sections 4 and 6 to wrap DuckDB connection in backoff to prevent race conditions when materializing all assets (Not an issue after these assets switch to using resources).

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

File: docs/dagster-university/pages/dagster-essentials/lesson-6/using-resources-in-assets.md [hunk 1]
```diff
@@ -111,6 +125,8 @@ To refactor `taxi_trips` to use the `database` resource, we had to:
        conn.execute(query)
    ```
 
+   Notice that we no longer need to use the `backoff` function. The Dagster `DuckDBResource` handles this functionality for us.
```
Comment (cmpadden) [inline docs/dagster-university/pages/dagster-essentials/lesson-6/using-resources-in-assets.md:128]: Nice, thanks for calling this out.

**Review by cmpadden — APPROVED** [review]
Great work! Be sure to links this or comment in the original GitHub issue once this is merged. Thanks!

File: docs/dagster-university/pages/dagster-essentials/lesson-4/loading-data-into-a-database.md [hunk 2]
```diff
@@ -54,7 +62,7 @@ Now that you have a query that produces an asset, let’s use Dagster to manage
...
 
-   4. A variable named `conn` is created, which defines the connection to the DuckDB database in the project. To do this, it uses the `.connect` method from the `duckdb` library, passing in the `DUCKDB_DATABASE` environment variable to tell DuckDB where the database is located.
+   4. A variable named `conn` is created, which defines the connection to the DuckDB database in the project. To do this, we first wrap everything with the Dagster utility function `backoff`. Using the backoff function ensures that multiple assets can use the DuckDB safely without locking resources. The backoff function takes in function we want to call, in this case the `.connect` method from the `duckdb` library, any errors to retry on (`RuntimeError` and `duckdb.IOException`), the max number of retires and finally the args to supply to the `.connect` DuckDB method. In this case we are passing in the `DUCKDB_DATABASE` environment variable to tell DuckDB where the database is located.
```
Comment (neverett) [inline docs/dagster-university/pages/dagster-essentials/lesson-4/loading-data-into-a-database.md:65]:
```suggestion
   4. A variable named `conn` is created, which defines the connection to the DuckDB database in the project. To do this, we first wrap everything with the Dagster utility function `backoff`. Using the backoff function ensures that multiple assets can use the DuckDB safely without locking resources. The backoff function takes in the function we want to call (in this case the `.connect` method from the `duckdb` library), any errors to retry on (`RuntimeError` and `duckdb.IOException`), the max number of retries, and finally, the arguments to supply to the `.connect` DuckDB method. In this case, we are passing in the `DUCKDB_DATABASE` environment variable to tell DuckDB where the database is located.
```

File: docs/dagster-university/pages/dagster-essentials/lesson-6/using-resources-in-assets.md (hunk 1 above)
Comment (neverett) [reply docs/dagster-university/pages/dagster-essentials/lesson-6/using-resources-in-assets.md:128]: seconded

**Review by neverett — APPROVED** [review]
Left some small copyediting suggestions, otherwise LGTM

**Pull request merged**